### DISCUSSION TITLE
Updating Referenced Packages and Models

### DISCUSSION THREAD

**Discussion by MatteoSchiavon** [OpenModelica/OpenModelica#12063] [Q&A] [answered]
Is there a smart way to update the name of a package? If for instance, the package doesn't save all its content in one file, all the models within that have the first line as "within -package path-" so the reference will break if the package name is changed. The same would happen for all the referenced package contents (in this latter case, the reference will break if we simply change the name of the single reference model/block).

**Comment by casella** [comment, marked as answer]
We need to implement a "Rename" function properly. It's on our to-do list.

**Reply by MatteoSchiavon** [reply]
We have done a very simple Python script to handle this, we could share it with you if you want to check it. In case it is properly working we could also contribute to implementing that in Modelica 😊

**Comment by AnHeuermann** [comment]
Or open a feature request in https://github.com/OpenModelica/modelica-language-server to get this working in VS Code.
It's probably not too hard to do this, but nobody is currently working on that.

**Reply by casella** [reply]
Sounds good. But we should get good Rename with refactoring in OMEdit as well, eventually.
